iced_layershell/src/multi_window/state.rs

```diff
@@ -41,62 +40,38 @@ where
         let theme = application.theme();
         let appearance = application.style(&theme);
 
-        let real_window_size = Size::new(width, height);
-        let viewport = Viewport::with_physical_size(
-            real_window_size,
-            wayland_scale_factor * application_scale_factor,
-        );
+        let logical_size = Size::new(width, height);
+        let viewport = viewport(logical_size, wayland_scale_factor, application_scale_factor);
+
+        let wpviewport = window
```

Comment:
At the beginning the size maybe 0,0 so I wait for the next time for the size being adjusted and set the destination

Reply:
The size should be provided by application. In which case will it be (0, 0)? Also, if the size has never been changed, the viewport will be wrong.

Reply:
for the iced_layershell to make a compositor, it need a waylandWrapper, at this time I cannot know the size, but when the surface is configured, it will update. so my design here is wait for the later update to update the viewport

Reply:
Sorry. I don't quite understand this case. I'm new to Wayland. I can't find `waylandWrapper` in this repo. Will this new code cause any issue in this situation?

Reply:
Emm.. what you have done is better.. but I think here , you should better to check if the size is (0,0 )

For example:

https://github.com/waycrate/exwlshelleventloop/blob/master/iced_layershell/src/multi_window.rs#L193

The window here is not configured, so the size is 0, 0, so this window need wait the next time that surface is configured to update the viewport

Reply:
Oh, I see. When I work with `daemon` to create a background process, it creates a fullscreen window even I don't send a `NewLayerShell` message. I think at this case the size will be (0, 0).

BTW, I want to create another PR. I think the default `StartMode` for daemon should be `Background`.

Back to this part, you mean I should check the size before `set_destination`? What happens if the application sends a `NewLayerShell` message with size (0, 0)?

Reply:
Background is just for background programs..and there are many programs. Because it does not use any wl_surface binded with the GUI, but is an extra one,  which used to create the composer, to avoid when gui is closed the composer also died. So this results in me needing to keep updating the surface , because it will never be received with an Outdate stated.  For a background program it may be enough, because it won't show for a long time.. but for a normal program, it consults a lot of resources.